tests/serverpod_test_server/test_integration/websockets/endpoint_websocket_connection_test.dart

```diff
@@ -1,75 +1,92 @@
 import 'package:serverpod_test_server/test_util/config.dart';
 import 'package:serverpod_test_server/test_util/test_serverpod.dart';
 import 'package:test/test.dart';
-import 'package:web_socket_channel/web_socket_channel.dart';
+import 'package:web_socket/web_socket.dart';
+import 'websocket_extensions.dart';
 
 void main() {
   group('Given endpoint websocket connection with connected client', () {
     var server = IntegrationTestServer.create();
-    late WebSocketChannel webSocket;
+    late WebSocket webSocket;
 
     setUp(() async {
       await server.start();
-      webSocket = WebSocketChannel.connect(
+      webSocket = await WebSocket.connect(
         Uri.parse(serverEndpointWebsocketUrl),
       );
-      await webSocket.ready;
     });
 
     tearDown(() async {
       await server.shutdown(exitProcess: false);
-      await webSocket.sink.close();
+      try {
+        await webSocket.close();
+      } on WebSocketConnectionClosed {
+        // Connection is already closed
+      }
     });
 
     test('when server is stopped then socket is closed.', () async {
-      webSocket.stream.listen((event) {
+      var isClosed = false;
+      webSocket.textEvents.listen((event) {
         // Listen to the stream to keep it open.
+      }, onDone: () {
+        isClosed = true;
       });
       // Await connection to be established and all handshakes to be done.
       await Future.delayed(Duration(seconds: 1));
 
       await server.shutdown(exitProcess: false);
-      expect(webSocket.closeCode, isNotNull);
+      expect(isClosed, isTrue);
     });
   });
 
   group('Given multiple endpoint websocket connections with connected clients',
       () {
     var server = IntegrationTestServer.create();
-    late WebSocketChannel webSocket1;
-    late WebSocketChannel webSocket2;
+    late WebSocket webSocket1;
+    late WebSocket webSocket2;
 
     setUp(() async {
       await server.start();
-      webSocket1 = WebSocketChannel.connect(
+      webSocket1 = await WebSocket.connect(
         Uri.parse(serverEndpointWebsocketUrl),
       );
-      await webSocket1.ready;
-      webSocket2 = WebSocketChannel.connect(
+      webSocket2 = await WebSocket.connect(
         Uri.parse(serverEndpointWebsocketUrl),
       );
-      await webSocket2.ready;
     });
 
     tearDown(() async {
       await server.shutdown(exitProcess: false);
-      await webSocket1.sink.close();
-      await webSocket2.sink.close();
+      try {
+        await Future.wait([
+          webSocket1.close(),
+          webSocket2.close(),
+        ]);
+      } on WebSocketConnectionClosed {
+        // Connections are already closed
+      }
```

Comment:
Again, if using `RelicWebSocket` on the client, this catch could be avoided with `tryClose()`.